apps/api/src/campaign-file/campaign-file.controller.ts

```diff
@@ -27,6 +28,10 @@ export class CampaignFileController {
 
   @Post(':campaign_id')
   @UseInterceptors(FilesInterceptor('file'))
+  @Roles({
+    roles: [RealmViewSupporters.role, ViewSupporters.role],
+    mode: RoleMatchingMode.ANY,
+  })
```

Comment:
Here it seems proper to allow campaign updates only to admins so let's leave it the way you've done it :waffle:

Reply:
My bad, it seems we actually need this functionality in order to attach files to a campaign during the campaign creation.

However, we need to restrict the upload only to user's own campaigns to avoid security leaks.